### PR TITLE
[Merged by Bors] - Docs for Bundle showing how to nest bundles

### PR DESCRIPTION
**Pull request by alice-i-cecile**
I've also added a clearer description of what bundles are used for, and explained that you can't query for bundles (a very common beginner confusion).

### CODE REVIEW

File: crates/bevy_ecs/src/bundle.rs [hunk 1]
```diff
@@ -8,13 +8,33 @@ use crate::{
...
 use std::{any::TypeId, collections::HashMap};
 
-/// An ordered collection of components
-///
-/// See [Bundle]
-/// # Safety
-/// [Bundle::type_info] must return the TypeInfo for each component type in the bundle, in the _exact_
-/// order that [Bundle::get_components] is called.
-/// [Bundle::from_components] must call `func` exactly once for each [TypeInfo] returned by [Bundle::type_info]
+/** An ordered collection of components, commonly used for spawning entities, and adding and removing components in bulk.
```
Comment (cart) [inline crates/bevy_ecs/src/bundle.rs:11]: We use `///`-style doc comments everywhere else. Can we do so here as well? Most other rust projects ive seen use that style too (including rustc)

File: crates/bevy_ecs/src/bundle.rs [hunk 2]
```diff
@@ -8,13 +8,33 @@ use crate::{
...
+
+You cannot query for a bundle, only individual components within it.
```
Comment (cart) [inline crates/bevy_ecs/src/bundle.rs:13]: Haha while we're touching this file, can you also add a note on "tuple bundles"?

File: crates/bevy_ecs/src/bundle.rs (hunk 1 above)
Comment (alice-i-cecile) [reply crates/bevy_ecs/src/bundle.rs:11]: Can do: I was just being lazy with it for the code example.

File: crates/bevy_ecs/src/bundle.rs [hunk 3]
```diff
@@ -8,13 +8,36 @@ use crate::{
...
 
-/// An ordered collection of components
+/// An ordered collection of components, commonly used for spawning entities, and adding and removing components in bulk.
 ///
-/// See [Bundle]
-/// # Safety
-/// [Bundle::type_info] must return the TypeInfo for each component type in the bundle, in the _exact_
-/// order that [Bundle::get_components] is called.
-/// [Bundle::from_components] must call `func` exactly once for each [TypeInfo] returned by [Bundle::type_info]
+///You cannot query for a bundle, only individual components within it.
```
Comment (mockersf) [inline crates/bevy_ecs/src/bundle.rs:13]: could you add a  ` `  between the `///` and the actual doc?

File: crates/bevy_ecs/src/bundle.rs [hunk 4]
```diff
@@ -8,13 +8,36 @@ use crate::{
...
+///
+///Typically, you will simply use `#[derive(Bundle)]` when creating your own `Bundle`.
+///The `Bundle` trait is automatically implemented for tuples of components:
+///`(ComponentA, ComponentB)` is a very convenient shorthand when working with one-off collections of components.
+///Note that both `()` and `(ComponentA, )` are valid tuples.
+///
+///You can nest bundles like so:
+///```
+///use bevy_ecs::bundle::Bundle;
```
Comment (mockersf) [inline crates/bevy_ecs/src/bundle.rs:22]: as this is not the import user should use, could you hide it ? with `/// # use...`

**Review by mockersf — CHANGES_REQUESTED** [review]

File: crates/bevy_ecs/src/bundle.rs (hunk 4 above)
Comment (alice-i-cecile) [reply crates/bevy_ecs/src/bundle.rs:22]: That's a useful trick. Added :)

**Comment by cart** [comment]
for future reference: rebasing instead of merging helps avoid pulling changes from `main` into the pr

**Pull request closed without merging**